modules/disk-uploader/pkg/secrets/secrets.go

```diff
@@ -13,29 +13,28 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-func CreateVirtualMachineExportSecret(k8sClient kubernetes.Interface, namespace, name string) error {
+func CreateVirtualMachineExportSecret(k8sClient kubernetes.Interface, namespace, name string) (*corev1.Secret, error) {
```

Comment:
What about letting this func take care of the naming completely instead of having the `name` argument? It might come as a surprise to callers that the name of the created object is different than what was passed in.

Reply:
Do you mean to have something like this: `GenerateName: "vmexport-secret-"`?

Reply:
I would vote for having at least some connection to the original source, because then you can easily identify which object can be checked or deleted, if eg. something gets wrong

Reply:
To identify created objects to delete you can take a look at the returned Secret, no?

Reply:
I'd consider modifying the name a surprising side effect of this method at least. If you want to keep this then I'd suggest to rename the argument to `baseName` or something.

Reply:
In the best case user should not care about these tmp objects, which are created just to achieve the upload of the disk. 
I agree the name param should be rename to something else.

Reply:
Done.